js/views/message_list_view.js

```diff
@@ -11,36 +11,43 @@
         itemView: Whisper.MessageView,
         events: {
             'scroll': 'onScroll',
-            'reset-scroll': 'resetScrollPosition'
         },
         onScroll: function() {
             this.measureScrollPosition();
             if (this.$el.scrollTop() === 0) {
                 this.$el.trigger('loadMore');
             }
+            if (this.bottomOffset === 0) {
+                this.$el.trigger('atBottom');
+            } else if (this.bottomOffset > this.outerHeight) {
+                this.$el.trigger('farFromBottom');
+            }
         },
         measureScrollPosition: function() {
             if (this.el.scrollHeight === 0) { // hidden
                 return;
             }
-            this.scrollPosition = this.$el.scrollTop() + this.$el.outerHeight();
+            this.outerHeight = this.$el.outerHeight();
+            this.scrollPosition = this.$el.scrollTop() + this.outerHeight;
             this.scrollHeight = this.el.scrollHeight;
             this.shouldStickToBottom = this.scrollPosition === this.scrollHeight;
             if (this.shouldStickToBottom) {
                 this.bottomOffset = 0;
             } else {
-                this.bottomOffset = this.scrollHeight - this.$el.scrollTop();
+                this.bottomOffset = this.scrollHeight - this.$el.scrollTop() - this.$el.outerHeight();
             }
         },
         resetScrollPosition: function() {
-            var scrollPosition = this.scrollPosition;
-            if (this.scrollHeight !== this.el.scrollHeight) {
-               scrollPosition = this.el.scrollHeight * this.scrollPosition / this.scrollHeight;
```

Comment:
Careful... iirc, this proportional scaling is intended to handle the case where the window height has changed in between measuring and resetting the scroll position.